Cabal-syntax/src/Distribution/Utils/Path.hs

```diff
@@ -531,3 +536,15 @@ data Response
 --
 -- See Note [Symbolic paths] in Distribution.Utils.Path.
 data PkgConf
+
+-------------------------------------------------------------------------------
+
+-- * Windows utils
+
+-------------------------------------------------------------------------------
+
+-- | Sometimes we need to represent a Windows path (that might have been
+-- normalized) as a POSIX path, for example in URIs.
```

Comment:
I don't see any added tests with this pull request. Would it be possible to add some? How about a doctest example attached to the haddocks of `asPosixPath`?

Reply:
Can you point me to a failed CI run or test before this fix?

Reply:
Would this be what you are looking for? https://github.com/haskell/cabal/actions/runs/12402807907/job/34625090748?pr=10647

Reply:
As you say, `asPosixPath` is trivial. How about a doctest example that shows what the problem is and how it was fixed? Could this be attached to `localToRemote` in `cabal-install/src/Distribution/Client/ProjectConfig/Legacy.hs`?

Reply:
I added a doctest in `asPosixPath` explaining why it would be used, referencing `network-uri`.